### PR TITLE
Fix Torch God biome torch swap issue

### PR DESCRIPTION
**Pull request by moisterrific**
Can confirm this fixes the issue but not sure if it opens any new exploits

?????? HAVE YOU UPDATED THE CHANGELOG? ??????
~not yet~ yes

Issue: https://github.com/Pryaxis/TShock/issues/2344

### CODE REVIEW

File: TShockAPI/Bouncer.cs [hunk 1]
```diff
@@ -303,11 +303,12 @@ internal void OnTileEdit(object sender, GetDataHandlers.TileEditEventArgs args)
...
-
-					if (selectedItem.placeStyle != style)
+					
+					if ((args.Player.TPlayer.BiomeTorchHoldStyle(style) != args.Player.TPlayer.BiomeTorchPlaceStyle(style))
+					&& (selectedItem.placeStyle != style))
 					{
-						TShock.Log.ConsoleError(string.Format("Bouncer / OnTileEdit rejected from (placestyle) {0} {1} {2} placeStyle: {3} expectedStyle: {4}",
-							args.Player.Name, action, editData, style, selectedItem.placeStyle));
+						TShock.Log.ConsoleError("Bouncer / OnTileEdit rejected from (placestyle) {0} {1} {2} placeStyle: {3} expectedStyle: {4}",
+						args.Player.Name, action, editData, style, selectedItem.placeStyle);
```
Comment (hakusaro) [inline TShockAPI/Bouncer.cs]:
```suggestion
							args.Player.Name, action, editData, style, selectedItem.placeStyle);
```

File: TShockAPI/Bouncer.cs [hunk 2]
```diff
@@ -303,11 +303,12 @@ internal void OnTileEdit(object sender, GetDataHandlers.TileEditEventArgs args)
 						args.Handled = true;
 						return;
 					}
-
-					if (selectedItem.placeStyle != style)
+					
+					if ((args.Player.TPlayer.BiomeTorchHoldStyle(style) != args.Player.TPlayer.BiomeTorchPlaceStyle(style))
+					&& (selectedItem.placeStyle != style))
```
Comment (hakusaro) [inline TShockAPI/Bouncer.cs:308]:
```suggestion
						&& (selectedItem.placeStyle != style))
```

**Review by hakusaro — CHANGES_REQUESTED** [review]
Thanks for your pull-request! Can you change the formatting so that you're indenting both the logical `and` et the arguments to `ConsoleError` so that they're easier to read?

**Comment by moisterrific** [comment]
> Thanks for your pull-request! Can you change the formatting so that you're indenting both the logical `and` et the arguments to `ConsoleError` so that they're easier to read?

sure thing

**Comment by hakusaro** [comment]
> sure thing

I actually fixed it for you but your fork isn't writable to us so I can't push it :S

**Review by hakusaro — APPROVED** [review]
lgtm_trick

**Pull request merged**